### PR TITLE
Allow to limit Poisson schedule with a stop time (closes #1617)

### LINKED ISSUE #1617: Add an optional stop time to the Poisson schedule
The `regular_schedule` can be configured with a start and stop time. Within this time window, events are generated. The `poisson_schedule` also allows for a start time but lacks a stop time.

Feature request: add an optional stop time to the Poisson schedule

### PR DESCRIPTION
**Pull request by schmitts**
Allow to limit Poisson schedule with a stop time using the same
semantics as the regular schedule.

### CODE REVIEW

**Review by noraabiakar — CHANGES_REQUESTED** [review]
Looks good thanks! I just have a small request to add an assertion.

**Review by noraabiakar — APPROVED** [review]
Thanks!

File: python/schedule.cpp [hunk 1]
```diff
@@ -256,6 +268,8 @@ void register_schedules(py::module& m) {
             "The expected frequency [kHz].")
         .def_readwrite("seed", &poisson_schedule_shim::seed,
             "The seed for the random number generator.")
+        .def_property("tstop", &poisson_schedule_shim::get_tstop, &poisson_schedule_shim::set_tstop,
+            "No events delivered after this time [ms].")
```
Comment (noraabiakar) [inline python/schedule.cpp:272]: `std::ostream& operator<<(std::ostream& o, const poisson_schedule_shim& p)` (line 27 of this file) needs to be amended to print out the `tstop` time as well.

Comment (schmitts) [reply python/schedule.cpp:272]: Will do.

**Pull request merged**